ElevatorSim/src/elevatorsim/floor/FloorController.java

```diff
@@ -46,6 +52,12 @@ public void run() {
 				server.sendElevatorRequest(request);
 				Thread.sleep(500);
 			}
+			
+			while(running) {
+				processEvents();
+				wait();
```

Comment:
what lock is this waiting on? don't use wait outside of synchronized blocks

Reply:
I was trying to use wait() to increase performance. I was thinking the thread should not be running when there is nothing inside the event queue. The thread is woken up via notifyAll() once a new event is added to the queue.

Reply:
I look into blockingqueue to see if it is better

Reply:
I will*

Reply:
why don't you just use Thread.sleep?